cmd/argo/commands/const.go

```diff
@@ -1,8 +1,61 @@
 package commands
 
-// Constants used by Argo
+import rbacv1 "k8s.io/api/rbac/v1"
+
 const (
-	ArgoServiceAccount = "argo"
-	ArgoClusterRole    = "argo-cluster-role"
-	ArgoServiceName    = "argo-ui"
+	// Argo controller resource constants
+	ArgoControllerServiceAccount     = "argo"
+	ArgoControllerClusterRole        = "argo-cluster-role"
+	ArgoControllerClusterRoleBinding = "argo-binding"
+
+	// Argo UI resource constants
+	ArgoUIServiceAccount     = "argo-ui"
+	ArgoUIClusterRole        = "argo-ui-cluster-role"
+	ArgoUIClusterRoleBinding = "argo-ui-binding"
+	ArgoUIDeploymentName     = "argo-ui"
+	ArgoUIServiceName        = "argo-ui"
+)
+
+var (
+	ArgoControllerPolicyRules = []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			// TODO(jesse): remove exec privileges when issue #499 is resolved
+			Resources: []string{"pods", "pods/exec"},
+			Verbs:     []string{"create", "get", "list", "watch", "update", "patch"},
+		},
+		{
+			APIGroups: []string{""},
+			Resources: []string{"configmaps"},
+			Verbs:     []string{"get", "watch", "list"},
+		},
+		{
+			APIGroups: []string{""},
+			Resources: []string{"persistentvolumeclaims"},
+			Verbs:     []string{"create", "delete"},
+		},
+		{
+			APIGroups: []string{"argoproj.io"},
+			Resources: []string{"workflows"},
+			Verbs:     []string{"get", "list", "watch", "update", "patch"},
+		},
+	}
+
+	ArgoUIPolicyRules = []rbacv1.PolicyRule{
```

Comment:
@jessesuen UI needs to create, get and update secrets. Same for config maps. Can you please update rule accordingly?

Reply:
Can you explain why argo UI needs to create/update secrets and configmaps? I though argo-ui was read only.

Reply:
Sorry, Argo CI needs it, not argo UI. LGTM